### PR TITLE
setuptools-scm: rework patches

### PR DESCRIPTION
**Pull request by lazka**
This patches setuptools-scm to make its git test suite pass.
Instead of converting absolute paths we just avoid them.

The upside here is that it is potentially upstreamable.. though not sure

### CODE REVIEW

**Comment by lazka** [comment]
@naveen521kk

**Comment by naveen521kk** [comment]
Possibly send this patch to upstream and see their response?

**Comment by lazka** [comment]
yeah, upstream still supports python2 so that wont work directly, after they are py3 only I'll try

**Comment by naveen521kk** [comment]
Let me try porting this to py2 and send them a PR.

File: mingw-w64-python-setuptools-scm/0001-git-Use-show-prefix-instead-of-show-toplevel-for-fin.patch [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
++        from pathlib import Path
++        import os
++        wd = os.path.abspath(wd)
++        real_wd, _, ret = do_ex("git rev-parse --show-prefix", wd)
+         if ret:
+             return
++        if not real_wd:
++            real_wd = wd
++        else:
++            real_dw = str(Path(wd).parents[len(Path(prefix).parents) - 1])
```
Comment (jeremyd2019) [inline mingw-w64-python-setuptools-scm/0001-git-Use-show-prefix-instead-of-show-toplevel-for-fin.patch:61]:
```suggestion
+            real_wd = str(Path(wd).parents[len(Path(prefix).parents) - 1])
```

Comment (lazka) [reply mingw-w64-python-setuptools-scm/0001-git-Use-show-prefix-instead-of-show-toplevel-for-fin.patch:61]: uh

Comment (lazka) [reply mingw-w64-python-setuptools-scm/0001-git-Use-show-prefix-instead-of-show-toplevel-for-fin.patch:61]: pushed to master

Comment (naveen521kk) [inline mingw-w64-python-setuptools-scm/0001-git-Use-show-prefix-instead-of-show-toplevel-for-fin.patch:61]: looks like prefix isn't defined no?

Comment (lazka) [reply mingw-w64-python-setuptools-scm/0001-git-Use-show-prefix-instead-of-show-toplevel-for-fin.patch:61]: yes. i changed that in master. thanks

**Pull request merged**